package.json

```diff
@@ -71,9 +71,10 @@
     "@types/semver": "7.3.9",
     "@types/unist": "2.0.6",
     "@typescript-eslint/eslint-plugin": "5.18.0",
+    "@typescript-eslint/parser": "5.18.0",
```

Comment:
I found the culprit: `@typescript-eslint/parser` must be explicitly installed, not sure why it wasn't before or the problem appeared after this Next.js upgrade.